test/src/e2e_vm_tests/test_programs/should_pass/stdlib/vec/src/main.sw

```diff
@@ -705,3 +708,127 @@ fn test_vector_with_capacity_u64() {
         Option::Some(val) => revert(0), Option::None => (), 
     }
 }
+
+fn test_vector_remove() {
+    let mut vector: Vec<u8> = ~Vec::new::<u8>();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+    let number2 = 2u8;
+    let number3 = 3u8;
+    let number4 = 4u8;
+
+    assert(vector.len() == 0);
+    assert(vector.capacity() == 0);
+    assert(vector.is_empty());
+
+    vector.push(number0);
+    vector.push(number1);
+    vector.push(number2);
+    vector.push(number3);
+    vector.push(number4);
+
+    assert(vector.len() == 5);
+    assert(vector.capacity() == 8);
+
+    // remove the first
+    let val = vector.remove(0);
+    assert(val == number0);
+    assert(vector.len() == 4);
+    assert(vector.capacity() == 8);
+
+    // remove the last
+    let val = vector.remove(3);
+    assert(val == number4);
+    assert(vector.len() == 3);
+    assert(vector.capacity() == 8);
+
+    // remove the middle
+    let val = vector.remove(1);
+    assert(val == number2);
+    assert(vector.len() == 2);
+    assert(vector.capacity() == 8);
+
+    match vector.get(0) {
+        Option::Some(val) => assert(val == number1), Option::None => revert(0), 
+    }
+
+    match vector.get(1) {
+        Option::Some(val) => assert(val == number3), Option::None => revert(0), 
+    }
+}
+
+fn test_vector_pop() {
+    let mut vector: Vec<u8> = ~Vec::new::<u8>();
+
+    let number0 = 0u8;
+    let number1 = 1u8;
+
+    vector.push(number0);
+    vector.push(number1);
+
+    match vector.pop() {
+        Option::Some(val) => assert(val == number1), Option::None => revert(0), 
+    }
+
+    match vector.pop() {
+        Option::Some(val) => assert(val == number0), Option::None => revert(0), 
+    }
+    
+    match vector.pop() {
+        Option::Some(val) => revert(0), Option::None => {}, 
```

Comment:
I'd feel a little better if we could assert that the last value returned is none rather than not reverting. if that's not possible in sway then perhaps it should be an issue?

Reply:
Not possible now, this requires to implement `Eq` for `Option<T>`, which is blocked by trait constraints